emissary-core/src/sam/parser.rs

```diff
@@ -319,6 +319,29 @@ impl<'a, R: Runtime> TryFrom<ParsedCommand<'a, R>> for SamCommand {
                     .and_then(|value| SamVersion::try_from(*value).ok()),
             }),
             ("SESSION", Some("CREATE")) => {
+                // checks if the inbound tunnel length is valid
+                if let Some(inbound_len) = value.key_value_pairs.get("inbound.length"){
+                    match inbound_len.parse::<u8>() {
+                        Ok(inbound_len) => {
+                            if inbound_len == 0 || inbound_len > 7 {
+                            return Err(());
```

Comment:
```suggestion
                            if inbound_len == 0 || inbound_len > 7 {
                            tracing::warn!(
                                target: LOG_TARGET,
                                ?inbound_len,
                                "invalid inbound tunnel length, 0-hop is not supported and 7 is the maximum length"
                            )
                            return Err(());
```
These actually should have warnings